Navbar

JoseLopezSolis opened the pull request:
We have created the "event" controller which is the controller to manage our events, also we created the navbar that shows a button to redirect the user to another view called "show", this view was created as well, and on this view, the user will be able to see the events that are stored in the database.

For more details see #2 
![Screenshot 2023-06-28 183023](https://github.com/BrightCoders-Institute/reto-administrador-de-eventos-jdg/assets/77039326/3d3b536a-4cb2-4200-89f9-bceed19bda6a)
![Screenshot 2023-06-28 182952](https://github.com/BrightCoders-Institute/reto-administrador-de-eventos-jdg/assets/77039326/10c6ba52-f014-4533-94b1-19728b6a5a97)

Commit:
We've created the navbar, the view show and the event model co-authored-by: hugocruz1234 cruzhugo1997@gmail.com co-authored-by: co-authored-by: ballo1988 rounderjards@gmail.com co-authored-by: sr-meh hector.m.rm@hotmail.com

cfcortesmx commented on app/assets/stylesheets/application.scss:
```diff
@@ -13,4 +13,20 @@
  */
- @import "bootstrap";
+@import "bootstrap";
+/* Base */
```
Consider adding a comment to explain what this section does.

cfcortesmx commented on app/assets/stylesheets/application.scss:
```diff
@@ -13,4 +13,20 @@
+    font-size: 25px;
+    font-weight: bold;
+  }
+  .hover-span:hover {
```
Consider adding a comment to explain what this section does.

cfcortesmx commented on app/assets/stylesheets/application.scss:
```diff
@@ -13,4 +13,20 @@
+  .hover-span:hover {
+    border-radius: 50%;
+  }
+  .hover-span:hover {
```
Consider adding a comment to explain what this section does.

cfcortesmx commented on app/controllers/events_controller.rb:
```diff
@@ -1,3 +1,10 @@
 class EventsController < ApplicationController
-  def index; end
-end
+  def index
```
Consider adding a comment to describe the purpose of the 'index' action.

cfcortesmx commented on app/controllers/events_controller.rb:
```diff
@@ -1,3 +1,10 @@
+    @events = Event.all
+  end
+
+  def show
```
Consider adding a comment to describe the purpose of the 'show' action.

Alonso2105 replied:
Checking the official rails documentation, the show method should be used to display the detail of an object, not the list of it, please consider following good practices andmaking this change, attached documentation
[Rails documentation](https://guides.rubyonrails.org/routing.html#crud-verbs-and-actions)

cfcortesmx commented on app/models/event.rb:
```diff
@@ -0,0 +1,2 @@
+class Event < ApplicationRecord
```
Consider adding some validations to the Event model to ensure data integrity.

cfcortesmx commented on app/views/events/index.html.erb:
```diff
@@ -1,5 +1,4 @@
-<div class="container">
-  <div class="row text-center">
+<div class="container d-flex justify-content-center">
```
Consider removing the unnecessary div tag.

cfcortesmx commented on app/views/events/index.html.erb:
```diff
@@ -1,5 +1,4 @@
-<div class="container">
-  <div class="row text-center">
+<div class="container d-flex justify-content-center">
     <h1>Events Management System</h1>
```
Consider removing the unnecessary div tag.

cfcortesmx commented on app/views/events/index.html.erb:
```diff
@@ -1,5 +1,4 @@
+<div class="container d-flex justify-content-center">
     <h1>Events Management System</h1>
-  </div>
 </div>
```
Consider removing the unnecessary div tag.

cfcortesmx commented on app/views/events/index.html.erb:
```diff
@@ -1,5 +1,4 @@
     <h1>Events Management System</h1>
-  </div>
 </div>
+
```
Consider removing the unnecessary div tag.

cfcortesmx commented on app/views/events/show.html.erb:
```diff
@@ -0,0 +1,47 @@
+<div class="container">
```
Consider adding a comment to describe the purpose of this file.

cfcortesmx commented on app/views/events/show.html.erb:
```diff
@@ -0,0 +1,47 @@
+<div class="container">
+  <h1>Events</h1>
```
Consider adding a comment to describe the purpose of this section.

cfcortesmx commented on app/views/events/show.html.erb:
```diff
@@ -0,0 +1,47 @@
+<div class="container">
+  <h1>Events</h1>
+
+<table class="table table-striped table-hover ">
```
Consider adding a comment to describe the purpose of this table.

cfcortesmx commented on app/views/events/show.html.erb:
```diff
@@ -0,0 +1,47 @@
+
+<table class="table table-striped table-hover ">
+  <thead class="table-dark"> 
+    <tr>
```
Consider adding a comment to describe the purpose of this table header.

cfcortesmx commented on app/views/events/show.html.erb:
```diff
@@ -0,0 +1,47 @@
+      <th scope="col">Date</th>
+      <th scope="col">Location</th>
+      <th scope="col">Cost</th>
+      <th colspan="2" scope="col">Actions</th>
```
Consider adding a comment to describe the purpose of this table header.

cfcortesmx commented on app/views/events/show.html.erb:
```diff
@@ -0,0 +1,47 @@
+    </tr>
+  </thead>
+  <tbody>
+    <% @events.each do |event| %>
```
Consider adding a comment to describe the purpose of this loop.

cfcortesmx commented on app/views/events/show.html.erb:
```diff
@@ -0,0 +1,47 @@
+          $<%= event.cost %>
+        </td>
+        <td>
+          <span class="material-symbols-outlined hover-span"> edit </span>
```
Consider using a button element instead of a span for better accessibility.

cfcortesmx commented on app/views/events/show.html.erb:
```diff
@@ -0,0 +1,47 @@
+          <span class="material-symbols-outlined hover-span"> edit </span>
+        </td>
+        <td>
+          <span class="material-symbols-outlined hover-span"> delete </span>
```
Consider using a button element instead of a span for better accessibility.

cfcortesmx commented on app/views/layouts/application.html.erb:
```diff
@@ -7,12 +7,13 @@
     <meta name="viewport" content="width=device-width,initial-scale=1">
     <%= csrf_meta_tags %>
     <%= csp_meta_tag %>
+    <link rel="stylesheet" href="https://fonts.googleapis.com/css2?family=Material+Symbols+Outlined:opsz,wght,FILL,GRAD@48,400,0,0" />
```
Consider moving the Google Fonts link to the head section of the HTML document for better performance.

cfcortesmx commented on app/views/navbar/_navbar.html.erb:
```diff
@@ -0,0 +1,11 @@
+<nav class="navbar navbar-expand-lg navbar-light bg-light">
```
Consider adding a comment to describe the purpose of this file.

cfcortesmx commented on app/views/navbar/_navbar.html.erb:
```diff
@@ -0,0 +1,11 @@
+<nav class="navbar navbar-expand-lg navbar-light bg-light">
+    <div class="container">
+      <a class="navbar-brand" href="/">Brighcoders Events</a>
```
Consider adding a comment to describe the purpose of this element.

cfcortesmx commented on app/views/navbar/_navbar.html.erb:
```diff
@@ -0,0 +1,11 @@
+        <span class="navbar-toggler-icon"></span>
+      </button>
+      <div class="collapse navbar-collapse justify-content-end" id="navbarNav">
+        <a class="btn btn-primary ms-lg-3" href="/show">Iniciar</a>
```
Consider adding a comment to describe the purpose of this link.

Alonso2105 replied:
please be consistent with the system, if it is in English or Spanish 👁️ and don't forget that your requirement is specific **[The menu should contain a button with the "Start" option.](https://github.com/BrightCoders-Institute/reto-administrador-de-eventos-jdg/issues/2#:~:text=The%20menu%20should%20contain%20a%20button%20with%20the%20%22Start%22%20option.)**

cfcortesmx commented on config/routes.rb:
```diff
@@ -1,3 +1,4 @@
 Rails.application.routes.draw do
-  get '/events', to: 'events#index'
+  get '/', to: 'events#index'
```
Consider using a more descriptive route name instead of '/'.

cfcortesmx commented on config/routes.rb:
```diff
@@ -1,3 +1,4 @@
 Rails.application.routes.draw do
-  get '/events', to: 'events#index'
+  get '/', to: 'events#index'
+  get '/show', to: 'events#show'
```
Consider using a more descriptive route name instead of '/show'.

Alonso2105 replied:
show method should return the detail of an object, check the official documentation for this modification please
[Rails Documentation](https://guides.rubyonrails.org/routing.html#connecting-urls-to-code)

cfcortesmx commented on test/models/event_test.rb:
```diff
@@ -0,0 +1,7 @@
+require "test_helper"
```
Consider adding a comment to describe the purpose of this file.

Alonso2105 commented on app/controllers/events_controller.rb:
```diff
@@ -1,3 +1,10 @@
+    @events = Event.all
+  end
+
+end
```
add end line please, if you are using `VScode` as editor you can leave this setting automatic on save
<img width="1440" alt="Screenshot 2023-06-28 at 22 52 49" src="https://github.com/BrightCoders-Institute/reto-administrador-de-eventos-jdg/assets/42948949/46fd9b39-c868-4f7a-be90-5bd2e2131e6d">

[EOL Documentation](https://instrumentationtools.com/use-end-line-eol-resistor-fire-gas-system/#:~:text=What%20are%20End%20of%20Line%20Resistors%20and%20do%20I%20need%20them%20on%20my%20alarm%20system%3F)

Alonso2105 commented on app/views/navbar/_navbar.html.erb:
```diff
@@ -0,0 +1,11 @@
+        <a class="btn btn-primary ms-lg-3" href="/show">Iniciar</a>
+      </div>
+    </div>
+</nav>
```
same here, add end line please

Alonso2105 commented on test/models/event_test.rb:
```diff
@@ -0,0 +1,7 @@
+require "test_helper"
+
+class EventTest < ActiveSupport::TestCase
+  # test "the truth" do
```
please add model tests
https://guides.rubyonrails.org/testing.html#rails-meets-minitest

JoseLopezSolis replied:
We're going to add the model test in the next issue, thanks for the observation.

juperameza commented on app/assets/stylesheets/application.scss:
```diff
@@ -13,4 +13,20 @@
+@import "bootstrap";
+/* Base */
+body {
+  padding: 0;
+  margin: 0;
```
Order this properties alphabetically.

juperameza replied:
And try to add some structure to the css for  better usage on the future

juperameza reviewed (changes requested):
Resolve Alonso conversations. And mine

Commit:
Recomendations fixed:
-controller name fixed
-SCSS desing fixed
-Adding a new line in each file
-Test file waiting for be modificated
-Button iniciar to Start
-Improved routes names

cfcortesmx commented on app/assets/stylesheets/application.scss:
```diff
@@ -13,4 +13,21 @@
  *= require_self
  */
- @import "bootstrap";
+//Importing bootstrap
```
Consider moving the import statement for bootstrap to the top of the file for better organization and readability.

cfcortesmx commented on app/assets/stylesheets/application.scss:
```diff
@@ -13,4 +13,21 @@
+//Importing bootstrap
+@import "bootstrap";
+/* Base */
+:root {
```
Consider using CSS variables for defining colors and font sizes. This will make it easier to update the styles in the future and maintain consistency throughout the application.

cfcortesmx replied:
There seems to be a typo in the variable name. It should be '--header-color' instead of '--head-color'.

cfcortesmx commented on app/assets/stylesheets/application.scss:
```diff
@@ -13,4 +13,21 @@
+body {
+  h1 {
+    color: var(--head-color);
+    font-size: var(--font-size-header);
```
Remove the commented out code. It's unnecessary and can clutter the codebase.

cfcortesmx commented on app/controllers/events_controller.rb:
```diff
@@ -1,3 +1,7 @@
 class EventsController < ApplicationController
-  def index; end
+  #This method will show the events in /events view
```
Consider renaming the method `events` to `index` to follow Rails conventions for controller actions.

cfcortesmx commented on app/controllers/events_controller.rb:
```diff
@@ -1,3 +1,7 @@
 class EventsController < ApplicationController
-  def index; end
+  #This method will show the events in /events view
+  def events
```
Remove the comment `#This method will show the events in /events view` as it is redundant and does not provide any additional information.

cfcortesmx commented on app/controllers/events_controller.rb:
```diff
@@ -1,3 +1,7 @@
+  def events
+    @events = Event.all
+  end
+
```
Remove the unused `show` action as it is not being used in this controller.

cfcortesmx commented on app/models/event.rb:
```diff
@@ -0,0 +1,7 @@
+class Event < ApplicationRecord
+  validates :title , presence:true
```
Consider adding a blank line after the class declaration for better readability.

cfcortesmx commented on app/models/event.rb:
```diff
@@ -0,0 +1,7 @@
+class Event < ApplicationRecord
+  validates :title , presence:true
+  validates :description , presence:true
```
Consider adding a blank line before the first validation for better readability.

cfcortesmx commented on app/models/event.rb:
```diff
@@ -0,0 +1,7 @@
+class Event < ApplicationRecord
+  validates :title , presence:true
+  validates :description , presence:true
+  validates :date , presence:true
```
Consider adding a blank line between each validation for better readability.

cfcortesmx commented on app/views/events/events.html.erb:
```diff
@@ -0,0 +1,48 @@
+    </tr>
+  </thead>
+  <tbody>
+  <!-- In this loop we're showing the events-->
```
Consider adding a comment to explain the purpose of this loop.

cfcortesmx commented on app/views/events/index.html.erb:
```diff
@@ -1,5 +1,3 @@
-<div class="container">
-  <div class="row text-center">
+<div class="container d-flex justify-content-center">
```
Consider removing the extra space at the end of the line.

cfcortesmx commented on app/views/layouts/application.html.erb:
```diff
@@ -3,6 +3,7 @@
   <head>
     <%= stylesheet_link_tag "application", "data-turbo-track": "reload" %>  
     <%= javascript_importmap_tags %>
+    <link rel="stylesheet" href="https://fonts.googleapis.com/css2?family=Material+Symbols+Outlined:opsz,wght,FILL,GRAD@48,400,0,0" />
```
Consider moving the Google Fonts stylesheet link to the head section of the HTML document for better performance. This will ensure that the font styles are loaded before rendering the page.

cfcortesmx commented on app/views/navbar/_navbar.html.erb:
```diff
@@ -0,0 +1,11 @@
+        <span class="navbar-toggler-icon"></span>
+      </button>
+      <div class="collapse navbar-collapse justify-content-end" id="navbarNav">
+        <a class="btn btn-primary ms-lg-3" href="/events">Start</a>
```
The button text should be more descriptive and meaningful. Instead of 'Start', consider using 'View Events' or something similar.

cfcortesmx commented on config/routes.rb:
```diff
@@ -1,3 +1,4 @@
 Rails.application.routes.draw do
-  get '/events', to: 'events#index'
+  root to: 'events#index'
```
Consider using the `root` method instead of `get '/'` to define the root route. This provides a more explicit and readable way to define the root route.

cfcortesmx commented on config/routes.rb:
```diff
@@ -1,3 +1,4 @@
 Rails.application.routes.draw do
-  get '/events', to: 'events#index'
+  root to: 'events#index'
+  get '/events', to: 'events#events'
```
The route '/show' seems unnecessary. If the purpose is to show events, it would be better to use a RESTful route like '/events/:id' to show a specific event. Consider removing this route and using a show action within the 'events' controller instead.

cfcortesmx commented on config/routes.rb:
```diff
@@ -1,3 +1,4 @@
-  get '/events', to: 'events#index'
+  root to: 'events#index'
+  get '/events', to: 'events#events'
 end
```
The route '/events' should be named more descriptively. Consider using a more meaningful name like '/events/:id' to show a specific event or '/events/new' to create a new event.

juperameza commented on app/controllers/events_controller.rb:
```diff
@@ -1,3 +1,7 @@
 class EventsController < ApplicationController
-  def index; end
+  #This method will show the events in /events view
+  def events
+    @events = Event.all
+  end
```
Why did you delete the restful methods?

JoseLopezSolis replied:
The index method didn't do anything, instead the events method does the request of the data, and show it in events view.

juperameza replied:
its better to do restful methods instead of that events its better to do index

Commit:
re-add the functions show and index on the event controller

juperameza commented on app/views/navbar/_navbar.html.erb:
```diff
@@ -5,7 +5,7 @@
       </button>
       <div class="collapse navbar-collapse justify-content-end" id="navbarNav">
-        <a class="btn btn-primary ms-lg-3" href="/events">Start</a>
+        <a class="btn btn-primary" href="/events/show">START</a>
```
Use link to in rails

hugocruz1234 committed:
Merge branch 'main' into navbar

JaimeGonz reviewed (approved):
Well done!
